### PR TITLE
fix(curriculum): include header in tests

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
- **fix(tests): include header as part of learner code**
- **test: add extra solution for personal portfolio**

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

https://github.com/freeCodeCamp/freeCodeCamp/pull/56570 highlighted a testing gap. The changes introduced in that PR made some valid project submissions fail the browser tests, but those same tests passed in CI.

This PR makes the CI tests use the preview header, same as the browser tests, and should catch these kinds of issues in the future.

### CODE REVIEW

File: curriculum/challenges/english/14-responsive-web-design-22/build-a-personal-portfolio-webpage-project/build-a-personal-portfolio-webpage.md [hunk 1]
```diff
@@ -280,3 +280,12 @@ p{
...
 ```
+
+---
+
+```html
+<head><style>@media (max-width: 500px){nav{display: none;}}</style></head><body><nav id="navbar"><a href="#projects">text</a> | </nav><main><section id="welcome-section"><h1>text</h1></section><hr><section id="projects"><h1>Projects</h1><h2 class="project-tile"><a id="profile-link" target="_blank" href="https://freecodecamp.org">text</a></h2></section><hr></body></html>
+```
+
+```css
+```
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/build-a-personal-portfolio-webpage-project/build-a-personal-portfolio-webpage.md:291]: What is this addition here doing?

Comment (ojeytonwilliams) [reply curriculum/challenges/english/14-responsive-web-design-22/build-a-personal-portfolio-webpage-project/build-a-personal-portfolio-webpage.md:291]: Most solutions for multifile challenges have separate CSS files. This
solution includes a style element in the html to better detect
interactions between the CSSHelper and the header.

Comment (ShaunSHamilton) [reply curriculum/challenges/english/14-responsive-web-design-22/build-a-personal-portfolio-webpage-project/build-a-personal-portfolio-webpage.md:291]: I am confused, though. This makes it look like there are two `index.html` files, and two `style.css` files for the solution? Which is not possible in the editor.

Comment (ojeytonwilliams) [reply curriculum/challenges/english/14-responsive-web-design-22/build-a-personal-portfolio-webpage-project/build-a-personal-portfolio-webpage.md:291]: Oh, I see. There are now two solutions, separated by `---`, both with 1 html and 1 css file.

Comment (ShaunSHamilton) [reply curriculum/challenges/english/14-responsive-web-design-22/build-a-personal-portfolio-webpage-project/build-a-personal-portfolio-webpage.md:291]: Oh. I did not know we had such a thing. Thanks 👍

Comment (ojeytonwilliams) [reply curriculum/challenges/english/14-responsive-web-design-22/build-a-personal-portfolio-webpage-project/build-a-personal-portfolio-webpage.md:291]: Yeah, it's a thing. We don't use it much and it's not exactly highlighted in the docs, but it exists.

**Review by naomi-lgbt — APPROVED** [review]

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍 

I think we might need to do work on the oddity of now having two `head` elements:

```html
<head></head><head><style>@media (max-width: 500px){nav{display: none;}}</style></head><body><nav id="navbar"><a href="#projects">text</a> | </nav><main><section id="welcome-section"><h1>text</h1></section><hr><section id="projects"><h1>Projects</h1><h2 class="project-tile"><a id="profile-link" target="_blank" href="https://freecodecamp.org">text</a></h2></section><hr></main></body><script id="fcc-test-runner" src='/js/frame-runner-543042f22803f54627ea.js' type='text/javascript'></script>
```

**Pull request merged**